### PR TITLE
Sophos Endpoint Protection: New App - First release

### PR DESCRIPTION
**Pull request by edro15**
#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [X] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
- [X] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [X] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes
First release for a Sophos Endpoint Protection app. Actions provided are listed below.

actions

### CODE REVIEW

**Comment by dgopani-crest** [comment]
Closing #1 as discussed in https://splunk.atlassian.net/browse/PAPP-21908 and considering this source code for initial release

**Comment by dgopani-crest** [comment]
Hi @edro15, we observed in the list endpoints action that `health_status`, `type`, and `lockdown_status` parameters have value_list and allow_list both. For Phantom, if we will use value_list for the parameter, the user can select only one value from the dropdown in that case allow_list will not be useful. 

As we don't have a working asset, we are not sure about the actual functionality. Can you please update the code and remove one of these based on the functionality?

**Comment by edro15** [comment]
Hi @dgopani-crest , docs cross-checked and code updated as requested. 🟢 😄

File: release_notes/unreleased.md [hunk 1]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Initial release with python 3
```
Comment (ngaglani-crest) [inline release_notes/unreleased.md:2]:
```suggestion
* Initial release as a community-supported app.
```

File: sophosendpointprotection.json [hunk 2]
```diff
@@ -0,0 +1,2207 @@
+{
```
Comment (ngaglani-crest) [inline sophosendpointprotection.json:1]: Can we change indentation to 4 tabs as almost all the apps are following the same standard?

File: sophosendpointprotection.json [hunk 3]
```diff
@@ -0,0 +1,2207 @@
...
+      "parameters": {
+        "page_from_key": {
+          "description": "The key of the item from where to fetch a page",
+          "data_type": "string",
+          "order": 0
+        },
+        "page_size": {
+          "description": "The size of the page requested",
+          "data_type": "string",
+          "default": 50,
```
Comment (ngaglani-crest) [inline sophosendpointprotection.json]: Datatype of default does not match with the specified data_type.

File: release_notes/unreleased.md (hunk 1 above)
Comment (dgopani-crest) [reply release_notes/unreleased.md:2]: Generally, we are not adding this
Screenshot 2021-12-10 at 2 45 14 PM

File: sophosendpointprotection.json [hunk 4]
```diff
@@ -0,0 +1,2207 @@
...
+          "data_type": "string",
+          "required": true,
+          "order": 0
+        }
+      },
+      "output": [
+        {
+          "data_path": "action_result.status",
+          "data_type": "string",
+          "column_name": "Status",
```
Comment (ngaglani-crest) [inline sophosendpointprotection.json:462]: Please remove this line as we are displaying `action_result.data.*.status` datapath in the table view.

File: sophosendpointprotection.json [hunk 5]
```diff
@@ -0,0 +1,2207 @@
...
+        "endpointid": {
+          "description": "The endpoint ID",
+          "data_type": "string",
+          "required": true,
+          "order": 0
+        },
+        "enabled": {
+          "description": "Whether Tamper Protection should be turned on for the endpoint",
+          "data_type": "boolean",
+          "default": "true",
```
Comment (ngaglani-crest) [inline sophosendpointprotection.json]:
```suggestion
          "default": true,
```

File: sophosendpointprotection.json [hunk 6]
```diff
@@ -0,0 +1,2207 @@
...
+          "order": 1
+        },
+        "regenerate_password": {
+          "description": "Whether a new Tamper Protection password should be generated",
+          "data_type": "boolean",
+          "default": "false",
```
Comment (ngaglani-crest) [inline sophosendpointprotection.json]:
```suggestion
          "default": false,
```

File: sophosendpointprotection_connector.py [hunk 7]
```diff
@@ -0,0 +1,829 @@
...
+    Get Partner Token.
+    """
+    # def _get_partner_token(self,jwt_token):
+    #     self.save_progress("Getting the partner token")
+    #     headers = {'Authorization': 'Bearer '.format(jwt_token)}
+    #     payload = {}
+    #     response = requests.request("POST", WHOAMI_ENDPOINT, headers=headers, data = payload)
+    #     return json.loads(response.text)['id']
+
+    """
```
Comment (ngaglani-crest) [inline sophosendpointprotection_connector.py]: Docstring should be moved below method definition.

File: sophosendpointprotection.json (hunk 2 above)
Comment (ebhoir-crest) [reply sophosendpointprotection.json:1]: yes updated indentation to 4 tabs.

File: sophosendpointprotection_connector.py (hunk 7 above)
Comment (ebhoir-crest) [reply sophosendpointprotection_connector.py]: Moved docstring below method definition.

File: readme.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
+<!-- File: readme.html
+  Copyright (c) 2018-2021 Splunk Inc.
```
Comment (dgopani-crest) [inline readme.html]: please correct copyright year

File: sophosendpointprotection.json (hunk 2 above)
Comment (ngaglani-crest) [reply sophosendpointprotection.json:1]: Sorry, for the confusion. I meant 4 spaces, instead of 2 spaces.

**Review by ngaglani-crest — APPROVED** [review]

**Review by joshquerinphantom — APPROVED** [review]

**Pull request merged**